jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -75,12 +75,15 @@
   private static final Duration LOGGING_THREAD_SHUTDOWN_TIMEOUT = Duration.ofSeconds(1);
 
   /** @return a GradleProjectProperties from the given project and logger. */
-  static GradleProjectProperties getForProject(Project project, Logger logger) {
-    return new GradleProjectProperties(project, logger);
+  static GradleProjectProperties getForProject(
+      Project project, Logger logger, TempDirectoryProvider tempDirectoryProvider) {
+    return new GradleProjectProperties(project, logger, tempDirectoryProvider);
   }
 
-  static Path getExplodedWarDirectory(Project project) {
-    return project.getBuildDir().toPath().resolve(ProjectProperties.EXPLODED_WAR_DIRECTORY_NAME);
+  String getWarFilePath() {
+    TaskProvider<Task> warTask = TaskCommon.getWarTaskProvider(project);
+    if (warTask != null) return warTask.get().getOutputs().getFiles().getAsPath();
+    else throw new GradleException("Obtaining project build output files failed");
```

Comment:
In this codebase we write all if statements using brackets, even one-liners. Plus the early return means we don't need the `else`, so these two lines can be rewritten as:
```
if (warTask != null) {
  return warTask.get().getOutputs().getFiles().getAsPath();
}
throw new GradleException("Obtaining project build output files failed");
```

That said, I think something like this would be even simpler:
```
Preconditions.checkNotNull(warTask, "Obtaining project build output files failed");
return warTask.get().getOutputs().getFiles().getAsPath();
```

Reply:
I tried the second suggestion and it lead to https://github.com/uber/NullAway/wiki/Error-Messages#dereferenced-expression-is-nullable error. So going with the first suggestion.